format/spec.md

```diff
@@ -167,30 +167,34 @@ A **`map`** is a collection of key-value pairs with a key type and a value type.
-| **`binary`**       | Arbitrary-length byte array                                              |                                                  |
+| Primitive type       | Description                                                              | Requirements                                     |
+|----------------------|--------------------------------------------------------------------------|--------------------------------------------------|
+| **`boolean`**        | True or false                                                            |                                                  |
+| **`int`**            | 32-bit signed integers                                                   | Can promote to `long`                            |
+| **`long`**           | 64-bit signed integers                                                   |                                                  |
+| **`float`**          | [32-bit IEEE 754](https://en.wikipedia.org/wiki/IEEE_754) floating point | Can promote to double                            |
+| **`double`**         | [64-bit IEEE 754](https://en.wikipedia.org/wiki/IEEE_754) floating point |                                                  |
+| **`decimal(P,S)`**   | Fixed-point decimal; precision P, scale S                                | Scale is fixed [1], precision must be 38 or less |
+| **`string`**         | Arbitrary-length character sequences                                     | Encoded with UTF-8 [2]                           |
+| **`uuid`**           | Universally unique identifiers                                           | Should use 16-byte fixed                         |
+| **`fixed(L)`**       | Fixed-length byte array of length L                                      |                                                  |
+| **`binary`**         | Arbitrary-length byte array                                              |                                                  |
+| **`date`**           | Calendar date, without time, without timezone                            | [3]                                              |
+| **`time`**           | Time of day, without date, without timezone                              | [3], [4]                                         |
+| **`timestamp`**      | Timestamp, with microsecond precision, without timezone                  | [4], [6]                                         |
+| **`timestamptz`**    | Timestamp, with microsecond precision, with timezone                     | [4], [7]                                         |
+| **`timestamp_ns`**   | Timestamp, with nanosecond precision, without timezone                   | [5], [6]                                         |
+| **`timestamptz_ns`** | Timestamp, with nanosecond precision, with timezone                      | [5], [7]                                         |
```

Comment:
We'll need to specify which types are allowed in different versions of the spec. All of the original types are allowed from v1 forward. The new types would need to be allowed from v3 forward.

Reply:
I've added this to Appendix E.